src/adapter/graphql/tests/tests/test_tasks.rs

```diff
@@ -150,10 +150,11 @@ async fn test_task_list_by_dataset() {
     let mut task_sched_mock = MockTaskScheduler::new();
     task_sched_mock
         .expect_list_tasks_by_dataset()
-        .return_once(move |_| {
-            Ok(Box::pin(futures::stream::iter(
-                [Ok(returned_task)].into_iter(),
-            )))
+        .return_once(move |_, _| {
+            Ok(TaskStateListing {
+                stream: Box::pin(futures::stream::iter([Ok(returned_task)].into_iter())),
```

Comment:
can try [`futures::stream::once`](https://docs.rs/futures/latest/futures/stream/fn.once.html)

Reply:
Slightly cleaner, but not much shorter:

`stream: Box::pin(futures::stream::once(async { Ok(returned_task) })),`